### PR TITLE
start services with docker-compose

### PR DESCRIPTION
**Pull request by graphaelli**
Adds `scripts/compose.py`, a utility for generating a `docker-compose.yml` useful for development and integration environments, and use it for integration tests under CI.

replaces #75

### CODE REVIEW

**Comment by graphaelli** [comment]
jenkins, test this please

**Comment by graphaelli** [comment]
jenkins, test this please

**Comment by graphaelli** [comment]
jenkins, test this please

**Comment by graphaelli** [comment]
Version tests pass on my machine but not on jenkins due to the max path length (`BINPRM_BUF_SIZE ` of 127) being exceeded under jenkins - `/var/lib/jenkins/workspace/elastic+apm-integration-testing+master+multijob-python-agent-versions/APM_SERVER/master/PYTHON_AGENT/github/bin/python`.  We'll have to figure that out before merging this.

File: README.md [hunk 1]
```diff
@@ -6,19 +6,109 @@ __WIP__
...
+
+### Running Tests
+
+All integration tests are written in python and live under `tests/`.
+
+Several `make` targets exist to make their execution simpler:
+
+- test-server
+- test-kibana
+- test-agent-{go,node,python,ruby,...}
```
Comment (jalvz) [inline README.md]: this still doesn't work for me, i only made work it once and never again :(

the issue is always a timeout on querying ES 

```
retry.py                   200 DEBUG    Converted retries value: False -> Retry(total=False, connect=None, read=None, redirect=0, status=None)
connectionpool.py          396 DEBUG    http://localhost:9200 "GET /apm-*/_count HTTP/1.1" 200 71
base.py                     83 INFO     GET http://localhost:9200/apm-*/_count [status:200 request:0.003s]
base.py                     85 DEBUG    > {"query": {"bool": {"must": [{"term": {"processor.event": {"value": "transaction"}}}]}}}
base.py                     86 DEBUG    < {"count":0,"_shards":{"total":0,"successful":0,"skipped":0,"failed":0}}
```

will have to spend some time tracking it down but would appreciate some help to see what i'm doing wrong

**Comment by graphaelli** [comment]
jenkins, test this please

**Comment by graphaelli** [comment]
failing on ci until container image with https://github.com/elastic/apm-server/pull/986 is built

**Comment by graphaelli** [comment]
jenkins, please test this

**Comment by graphaelli** [comment]
jenkins, test this please

File: README.md (hunk 1 above)
Comment (graphaelli) [reply README.md]: there's an issue with the current snapshots - let's rerun when something newer than jun-1 is available.

**Comment by graphaelli** [comment]
jenkins, retest this please

**Pull request merged**